maptivist/src/main/webapp/filter.js

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Filters markers based on category
+ */
+
+function getMarkersByCategory() {
+    const category = document.getElementById("category");
+    fetch("/marker").then(response => response.json()).then((markers) => {
+        console.log(markers);
```

Comment:
Beware too many console.log statements. It may slow things down and too much information may make debugging even more difficult.